### ISSUE TITLE
`__module__` is set wrongly when using class definition syntax

### ISSUE THREAD

**Issue by carljm** [python/mypy_extensions#2]
```
$ cat foo.py
from mypy_extensions import TypedDict


class Person(TypedDict):
    name: str

Host = TypedDict("Host", {"name": "str"})

$ python
Python 3.6.5 (default, Jun 17 2018, 12:13:06)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import foo
>>> foo.Host
<class 'foo.Host'>
>>> foo.Host.__module__
'foo'
>>> foo.Person
<class '_frozen_importlib.Person'>
>>> foo.Person.__module__
'_frozen_importlib'
>>>
```

Expected behavior is that `__module__` should be set correctly when either definition syntax is used.

I haven't experimented yet, but I expect that with the class definition syntax, we don't need a manual frame-walking override of `__module__` at all, since Python will set it correctly for us.

**Comment by ilevkivskyi** [comment]
This looks like a `sys._getframe()` bug, because class syntax a function syntax have different call nesting.

Would you like to attempt a PR?

**Comment by carljm** [comment]
> This looks like a sys._getframe() bug, because class syntax a function syntax have different call nesting.

Yes, although I suspect the simpler fix may be to skip manually setting `__module__` at all when using the class-definition syntax, since Python should automatically set it correctly.

> Would you like to attempt a PR?

Yes, I or someone on my team will submit a PR for this shortly.

**Comment by msullivan** [comment]
Fixed by #4
